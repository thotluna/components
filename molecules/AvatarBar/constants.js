export const options = {
  sizes: ['sm', 'lg'],
}
