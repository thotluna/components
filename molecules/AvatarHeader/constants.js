export const options = {
  size: ['sm', 'md'],
}
