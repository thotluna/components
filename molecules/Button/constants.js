export const options = {
  colors: ['base', 'primary', 'secondary'],
}
