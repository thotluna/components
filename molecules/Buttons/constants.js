export const options = {
  size: ['full', 'short', 'mini'],
  color: ['primary', 'secondary', 'inverted'],
}
