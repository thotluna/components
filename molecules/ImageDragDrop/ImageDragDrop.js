import React, { useRef, useState } from 'react'
import PropTypes from 'prop-types'
import { styles } from '.'
import { getClasses } from '../../helpers/styles'

import Paragraph from '../../atoms/Paragraph'

import { usePreviewImage } from '../../hooks/useImagePreview'

const DRAG_IMAGE_STATES = {
  NONE: false,
  DRAG_OVER: true,
}

export const ImageDragDrop = ({ width, handelFile }) => {
  const inputRef = useRef()
  const [drag, setDrag] = useState(DRAG_IMAGE_STATES.NONE)
  const { file, setFile, imgURL, loading } = usePreviewImage(
    `https://dummyimage.com/${width}x${width}/#b4d1f/#3b3f4`
  )

  const handleDragEnter = (e) => {
    e.preventDefault()
    setDrag(DRAG_IMAGE_STATES.DRAG_OVER)
  }
  const handleDragLeave = (e) => {
    e.preventDefault()
    setDrag(DRAG_IMAGE_STATES.NONE)
  }
  const handleOnDrop = (e) => {
    e.preventDefault()
    setDrag(DRAG_IMAGE_STATES.NONE)
    const upFile = e.target.files[0]
    console.log(upFile)
    if (upFile) {
      setFile(upFile)
      handelFile(file)
    }
  }

  const getStyles = getClasses(styles)({})
  return (
    <div
      style={{ width, height: width }}
      className={getStyles('imageDragDrop', { 'is-drag': drag })}
      onClick={() => {
        inputRef.current.click()
      }}
      onKeyPress={(e) => {
        if (e.code === 'Enter') {
          inputRef.current.click()
        }
      }}
      tabIndex="0"
    >
      {loading && <Paragraph>Loading...</Paragraph>}
      <img
        style={{ width: width, height: width }}
        src={imgURL}
        alt="insert your photo"
        onDragEnter={handleDragEnter}
        onDragLeave={handleDragLeave}
        onDragOver={handleDragEnter}
        onDrop={handleOnDrop}
      />
      <input
        className={getStyles('input-image')}
        type="file"
        accept="image/*"
        ref={inputRef}
        onChange={handleOnDrop}
      />
    </div>
  )
}

ImageDragDrop.propTypes = {
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  handelFile: PropTypes.func,
}

ImageDragDrop.defaultProps = {}

export default ImageDragDrop
