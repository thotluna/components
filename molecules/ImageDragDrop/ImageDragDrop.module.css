.imageDragDrop {
  display: flex;
  align-items: center;
  justify-content: center;
  border: 4px dashed var(--color-basic-white);
  border-radius: var(--border-radius-full);
}

.is-drag {
  border: 4px dashed var(--color-primary);
}

.imageDragDrop > img {
  border-radius: var(--border-radius-full);
  object-fit: cover;
}

.input-image {
  display: none;
}
