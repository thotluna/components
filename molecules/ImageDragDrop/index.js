export { default } from './ImageDragDrop'
export { options } from './constants'
export { default as styles } from './ImageDragDrop.module.css'
