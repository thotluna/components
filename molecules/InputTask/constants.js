export const status = {
  STATUS_STANDBY: 'standby',
  STATUS_CREATING: 'creating',
}

export const options = {
  status: [status.STATUS_STANDBY, status.STATUS_CREATING],
}
