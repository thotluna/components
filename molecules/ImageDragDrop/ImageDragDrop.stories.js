import React from 'react'
import ImageDragDrop from './ImageDragDrop'
import { options } from './constants'
import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <ImageDragDrop {...args} />
const ListTemplate = getVariusTemplate(ImageDragDrop)

export default {
  title: 'Molecules/ImageDragDrop',
  component: ImageDragDrop,
  args: {
    width: 150,
    handelFile: () => {},
  },
  argTypes: {
    children: { control: 'text' },
    color: getOptionsArgTypes(options.colors),
  },
}

export const Default = Template.bind({})

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }
