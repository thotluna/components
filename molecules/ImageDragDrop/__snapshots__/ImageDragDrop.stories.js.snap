// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/ImageDragDrop Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="imageDragDrop"
    onClick={[Function]}
    onKeyPress={[Function]}
    style={
      Object {
        "height": 150,
        "width": 150,
      }
    }
    tabIndex="0"
  >
    <img
      alt="insert your photo"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      src="https://dummyimage.com/150x150/#b4d1f/#3b3f4"
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
    <input
      accept="image/*"
      className="input-image"
      onChange={[Function]}
      type="file"
    />
  </div>
  <div
    className="imageDragDrop"
    onClick={[Function]}
    onKeyPress={[Function]}
    style={
      Object {
        "height": 150,
        "width": 150,
      }
    }
    tabIndex="0"
  >
    <img
      alt="insert your photo"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      src="https://dummyimage.com/150x150/#b4d1f/#3b3f4"
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
    <input
      accept="image/*"
      className="input-image"
      onChange={[Function]}
      type="file"
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/ImageDragDrop Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="imageDragDrop"
    onClick={[Function]}
    onKeyPress={[Function]}
    style={
      Object {
        "height": 150,
        "width": 150,
      }
    }
    tabIndex="0"
  >
    <img
      alt="insert your photo"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      src="https://dummyimage.com/150x150/#b4d1f/#3b3f4"
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
    <input
      accept="image/*"
      className="input-image"
      onChange={[Function]}
      type="file"
    />
  </div>
</div>
`;
